core/src/main/java/ai/timefold/solver/core/config/solver/SolverConfig.java

```diff
@@ -432,6 +443,14 @@ public void setMonitoringConfig(@Nullable MonitoringConfig monitoringConfig) {
     // With methods
     // ************************************************************************
 
+    public @NonNull SolverConfig withPreviewFeature(@NonNull PreviewFeature previewFeature) {
```

Comment:
I'd consider making this a vararg, and override the list every time.
If nothing is provided, nothing is enabled.
If something is provided, then that's what's enabled.